php-classes/Validators.class.php

```diff
@@ -56,10 +56,15 @@ public static function number($number, $options = array())
             && (($options['max'] === false) || ($number <= $options['max']));
     }
 
-    public static function email($email)
+    public static function email($email, $options = [])
     {
+        $options = array_merge([
+            'domain' => null
+        ], $options);
+
         return !empty($email)
-            && preg_match('/^[_a-zA-Z0-9-+]+(\.[_+a-zA-Z0-9-]+)*@[a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)*(\.[a-zA-Z]{2,})$/', $email);
+            && preg_match('/^[_a-zA-Z0-9-+]+(\.[_+a-zA-Z0-9-]+)*@[a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)*(\.[a-zA-Z]{2,})$/', $email)
+            && (empty($options['domain']) || preg_match('/^[_a-zA-Z0-9-+]+(\.[_+a-zA-Z0-9-]+)*@'.$options['domain'].'$/', $email));
```

Comment:
An example of why `preg_quote` is needed: If `"example.com"` was provided for domain, `"example!com"` would have passed